Description: Support architecture with align(double) < sizeof(int)
 Fontconfig checks to make sure the computed machine characteristics
 match what the compiler actually generates for data structure sizes.
 That check was wrong when the alignment requirements for double
 were smaller than the size of int. The fix for this that was merged
 upstream has been integrated into this Debian package.
 .
 fontconfig (2.9.0-3) unstable; urgency=low
 .
   * Apply upstream fix for FTBFS on m68k. Closes: #671072.
Author: Keith Packard <keithp@debian.org>
Bug-Debian: http://bugs.debian.org/671072

---

Origin: upstream
Bug-Debian: http://bugs.debian.org/671072
Reviewed-By: Akira TAGOH <akira@tagoh.org>
Last-Update: 2012-05-01

--- a/src/fcarch.c
+++ b/src/fcarch.c
@@ -56,7 +56,9 @@
 FC_ASSERT_STATIC (SIZEOF_VOID_P == sizeof (FcCharLeaf **));
 FC_ASSERT_STATIC (SIZEOF_VOID_P == sizeof (FcChar16 *));
 
-FC_ASSERT_STATIC (0x08 + 1*ALIGNOF_DOUBLE == sizeof (FcValue));
+#define FC_MAX(a,b)	((a) > (b) ? (a) : (b))
+
+FC_ASSERT_STATIC (0x08 + 1*FC_MAX(4,ALIGNOF_DOUBLE) == sizeof (FcValue));
 FC_ASSERT_STATIC (0x00 + 2*SIZEOF_VOID_P == sizeof (FcPatternElt));
 FC_ASSERT_STATIC (0x08 + 2*SIZEOF_VOID_P == sizeof (FcPattern));
 FC_ASSERT_STATIC (0x08 + 2*SIZEOF_VOID_P == sizeof (FcCharSet));
